### ISSUE TITLE
Catalog Answers not allowing not-strings

### ISSUE THREAD

**Issue by wyattjsmith1** [rancher/rancher#14711]
I at trying to start a Helm chart (gitlab). I would like to add an answer with a boolean. I have added it in the ui like so:
<img width="1150" alt="input" src="https://user-images.githubusercontent.com/17460119/43112351-83a08cfe-8eaa-11e8-8c97-a2a06d9fdc80.png">

The resulting json is a literal string `"false"`. Here is a screenshot of this specific field. To make it more concise, I eliminated other fields from the result.
<img width="368" alt="result" src="https://user-images.githubusercontent.com/17460119/43112382-b83f779a-8eaa-11e8-9d9b-08e66b11f512.png">

Is there any way to specify a non-string value in an answer?

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.0.6` UI: `v2.0.61` |
|Access|`local` `admin`|
|Route|`catalog-tab.launch`|

**Comment by vincent99** [comment]
Answers/helm variables are always strings.

**Comment by wyattjsmith1** [comment]
@vincent99 I'm not sure that is true. According to Helm's site, Helm charts allow non-string values: https://docs.helm.sh/chart_best_practices/#make-types-clear. If Rancher answers doesn't allow non strings, it looks like this could make using some helm charts difficult. You are obviously more knowledgeable than me on this subject, but it looks like Helm allows non strings.

**Comment by vincent99** [comment]
We use `--set-string`

https://github.com/rancher/rancher/blob/51d8300cf401825b3d06c3a7a8316d5909deac1b/pkg/controllers/user/helm/common.go#L181-L192

because it is predictable; `--set` does automatic type inference and causes seemingly (to the user) random problems depending on the value they provide (e.g. https://github.com/helm/helm/issues/1707)

**Comment by wyattjsmith1** [comment]
Ok, fair enough. Is there any way to force a non-string? Right now, the Gitlab Helm chart has fields that require non string values: https://gitlab.com/charts/gitlab/blob/master/doc/installation/command-line-options.md

Configuring Gitlab seems to be impossible with Rancher 2 if there is no way to specify a non string value. Is that correct?

**Comment by wyattjsmith1** [comment]
@vincent99 I opened a ticket for this on the Gitlab team, and they have expressed concerns about non string values across many charts. Is there any way to go around this?

https://gitlab.com/charts/gitlab/issues/616

**Comment by wyattjsmith1** [comment]
Would creating a questions.yaml help with this issue? This ticket talks about quotes on boolean values #13843. Has that been changed to force booleans to be strings as well?

**Comment by stieler-it** [comment]
Having the same problem as @wyattjsmith1 I have just deployed the "master" Docker image. Unfortunately I don't see any difference. Value certmanager.install = false leads to "certmanager.install":"false" in answers.yaml (looks more like JSON btw) and also the same error message. Docker master was pushed yesterday - should it contain the fix already?

**Comment by vincent99** [comment]
All that was done so far is reverting using set-string back to set, which was only in 2.0.7-rcX and never shipped in a release.  

The input to helm is still all strings and they have to do type guessing, it doesn't matter whether the yaml says  `false` or `"false"`* because both result in `heml … --set x=false`. (And quoted keys are perfectly valid yaml, as is anything that's valid json because yaml is a strict superset of yaml. The  answers "file" isn't actually sent anthere anyway, it's just for display and the answers are sent with the JSON in the POST/PUT request.)

So it would seem boolean answers have never worked, and don't work in helm.  Answers/set isn't the same as editing values.yaml.

**Comment by wyattjsmith1** [comment]
@vincent99 I am starting to think that this ticket and #14608 are the same bug with different symptoms. I have noticed that some parameters are not set for some reason, though there seems to be no logic to it. Here is a chart of working and not working parameters. I derived this by adding or removing these values to see if there is a change in the output:
- Working
  - `gitlab-runner.install` (Boolean)
- Not Working
  - `global.ingress.configureCertmanager` (Boolean)
  - `certmanager-issuer.email` (String)

If you look at the other ticket, people are posting other parameters that aren't working. I originally thought it had to do with subcharts, but it looks like `gitlab-runner.install` is working.
